### PR TITLE
[react-slider] Update react-slider types to 1.1

### PR DESCRIPTION
**Pull request by loichuder**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: see https://github.com/zillow/react-slider/blob/master/CHANGELOG.md#110-2020-11-03 and https://zillow.github.io/react-slider/
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

The full diff is messy as I reorganized the types to match the order of https://zillow.github.io/react-slider/ in a separate commit. The "real" update to 1.1 is in the first commit.

### CODE REVIEW

File: types/react-slider/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+// Type definitions for react-slider 1.1
 // Project: https://github.com/zillow/react-slider
 // Definitions by: Jason Unger <https://github.com/jsonunger>
 //                 Björgvin Bæhrenz Þórðarson <https://github.com/bjorgvin>
+//                 Loïc Huder <https://github.com/loichuder>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
 import * as React from 'react';
-import * as motion from 'react-motion';
```
Comment (loichuder) [inline types/react-slider/index.d.ts:9]: Not sure what this was doing here

File: types/react-slider/index.d.ts [hunk 2]
```diff
@@ -124,19 +124,42 @@ declare namespace ReactSlider {
...
-        ariaValuetext?: string | ((value: {index: number, value: number | number[], valueNow: number }) => string);
+        ariaValuetext?: string | ((value: { index: number; value: number | number[]; valueNow: number }) => string);
         /**
          * Provide a custom render function for the track node.
          * The render function will be passed two arguments. The first is
          * an object that should be added to your handle node.
          */
-        renderTrack?: (props: object, state: {index: number, value: number | number[] }) => JSX.Element;
+        renderTrack?: (
+            props: React.HTMLProps<HTMLDivElement>,
```
Comment (loichuder) [inline types/react-slider/index.d.ts:149]: Strengthened the typings of `props` in all `render` methods

File: types/react-slider/index.d.ts [hunk 3]
```diff
@@ -124,19 +124,42 @@ declare namespace ReactSlider {
...
+         * The CSS class set on the marks
+         */
+        markClassName?: string;
+        /**
+         * Provide a custom render function for the mark node.
+         * The render function will be passed one argument,
+         * an object with props that should be added to your handle node.
          */
-        renderThumb?: (props: object, state: {index: number, value: number | number[], valueNow: number }) => JSX.Element;
+        renderMark?: (props: React.HTMLProps<HTMLSpanElement>) => JSX.Element;
```
Comment (loichuder) [inline types/react-slider/index.d.ts]: Adding types for `marks` and the related functions added in 1.1

File: types/react-slider/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/react-slider/index.d.ts:9]: seems to be there from the very beginning of time:
https://github.com/loichuder/DefinitelyTyped/commit/917cb109d0cc654588357ccf3171648d3e10bcd1
current package has no dependency to that one, let's drop

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@loichuder thx!

**Comment by loichuder** [comment]
Ready to merge

**Pull request merged**